.github/workflows/sync-opencraft-forks-with-upstream.yml

```diff
@@ -45,7 +45,7 @@ jobs:
         run: git remote add upstream ${{ matrix.upstream_repo }}
 
       - name: Fetch Latest from Upstream
-        run: git fetch upstream
+        run: git fetch upstream ${{ matrix.upstream_branch }}
```

Comment:
@yusuf-musleh, [it looks like](https://github.com/open-craft/.github/actions/runs/5584426513/jobs/10205843352) this change didn't improve the fetching time. How about we change the "Checkout Forked Repo" step to check out the upstream branch instead? It would look line this:

```yaml
- name: Checkout upstream repo
  uses: actions/checkout@v3
  with:
    repository: ${{ matrix.upstream_repo }}  # We may need to change this variable to `openedx/edx-platform`
    ref: ${{ matrix.upstream_branch }}
```

Then, we would do something like:

```bash
git remote set-url --push origin https://${{ env.GITHUB_TOKEN }}@github.com/${{ matrix.fork_repo }}
```

We could also simplify the "Create a new branch off Upstream master" step by removing the last parameter, as the upstream branch would be our `origin`.

What do you think?

Reply:
That's a great idea! I updated the implementation in this PR and [ran the action](https://github.com/open-craft/.github/actions/runs/5585316590/jobs/10207894877), it now only takes a few seconds to fetch the upstream branch.